File: LeetCode/Medium/0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        l, r = 0, int(c ** 0.5)
        
        while l <= r:
            temp = l * l + r * r

            if temp == c:
                return True
            elif temp < c:
                l += 1
            else:
                r -= 1
        
        return False

File: LeetCode/Medium/0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long long l = 0;
        long long r = ceil(sqrt(c));

        while (l <= r){
            long long temp = l * l + r * r;

            if (temp == c) return true;
            else if (temp < c) ++l;
            else r--;
        }

        return false;
    }
};